tests/test_ote_api.py

```diff
@@ -331,19 +331,20 @@ def eval(task: OTEDetectionTask, model: Model, dataset: Dataset) -> Performance:
         performance = task.evaluate(result_set)
         return performance
 
-    def train_and_eval(self, template_dir):
-        """
-        Run training, analysis, evaluation and model optimization
+    def check_threshold(self, reference, value, delta_tolerance, message=''):
+        delta = value.score.value - reference.score.value
+        self.assertLessEqual(
+            np.abs(delta),
+            delta_tolerance,
+            msg=message +
+                f' (reference metric: {reference.score.value}, '
+                f'actual value: {value.score.value}, '
+                f'delta tolerance threshold: {delta_tolerance})'
+            )
+
+    def end_to_end(self, template_dir, quality_score_threshold=0.5, reload_perf_delta_tolerance=0.0,
+        export_perf_delta_tolerance=0.0005, pot_perf_delta_tolerance=0.1):
 
-        Flow of the test:
-        - Creates a randomly annotated project with a small dataset containing 3 classes:
-            ['rectangle', 'triangle', 'circle'].
-        - Trains a model for 10 epochs. Asserts that validation F-measure is larger than the threshold and
-            also that OpenVINO optimization runs successfully.
-        - Reloads the model in the task and recompute the performance. Asserts that the performance
-            difference between the original and the reloaded model is smaller than 1e-4. Ideally there should be no
-            difference at all.
-        """
         hyper_parameters, model_template = self.setup_configurable_parameters(template_dir, num_iters=150)
         detection_environment, dataset = self.init_environment(hyper_parameters, model_template, 250)
```

Comment:
If you are checking accuracy on this test, I would suggest making the test behavior deterministic. I mean fixed dataset at least and maybe to increase dataset to obtain more stable results.

Reply:
This is a reasonable thing to do. Meanwhile, what is the preferred strategy for POT tests? Are we going to disable them completely or just relax the threshold and hope, that it will be enough?

Reply:
I think the strategy is to optimize the model with the expected accuracy using POT at the end of the pipeline. The POT is the deterministic tool and depends on the dataset and input model. So If the test will be deterministic (I mean the dataset will be fixed and training will be deterministic too) then we can adjust the threshold for POT (it is necessary because the validation dataset is very small).

Reply:
Please choose one of the two options listed above as a temporary solution.

Reply:
Ok, It's better to relax the threshold. Could you file the issue by this discussion?